Add report card lost or damaged

JKobrynski opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ #22879 
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Add the following code to **InitialSettingsPage.js**

```js
useEffect(() => {
    window.Onyx.merge(`cardList`, {
        virtual: {
            cardID: 234523452345,
            state: 3,
            bank: 'Expensify',
            availableSpend: 10000,
            domainName: 'Expensify',
            lastFourPAN: '',
            isVirtual: true,
            cardholderFirstName: 'Test',
            cardholderLastName: 'Test',
        },
        physical: {
            cardID: 234523452345,
            state: 3,
            bank: '1000',
            availableSpend: 10000,
            domainName: 'Expensify',
            lastFourPAN: '2345',
            isVirtual: false,
        },
    });

    Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute('expensify'));
}, []);
```

2. Open settings, you should be automatically navigated to the cards page
3. Click the "Report physical card loss / damage" button
4. You should be navigated to a new page that will allow you to select the reason
5. Click the "Next" button
6. You should see a button with your address, if you haven't added your address yet, you can press it to do so
7. On submit, it should load for a second and then navigate back. If there is no address, an error should be displayed.

(Check recordings, they explain a lot)

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
9. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/44364829/b433f179-52d1-4077-bc5d-4df215fd691d
https://github.com/Expensify/App/assets/44364829/7c409b2a-b378-487d-af12-0f952759d386

</details>

<details>
<summary>Mobile Web - Chrome</summary>

[chrome1-compressed.webm](https://github.com/Expensify/App/assets/44364829/2493713d-be57-445b-889f-87ea52b64ed2)
[chrome2-compressed.webm](https://github.com/Expensify/App/assets/44364829/c075e29c-f491-4491-a29e-1d903bcdaf72)

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/44364829/1807dd43-2d6c-40ae-a8d4-84c7703eda99
https://github.com/Expensify/App/assets/44364829/28805647-4615-4158-8f9e-214be8492f2c

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/44364829/619afb13-0444-407d-85cc-12fb5ae21ae2
https://github.com/Expensify/App/assets/44364829/2e8af130-ef20-47c7-a1a4-46dc4f5487c0

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/44364829/c7efda39-3dc0-49bd-a8c4-38309cc3706d
https://github.com/Expensify/App/assets/44364829/fc537680-e3ba-4faf-838e-780c68f3c80f

</details>

<details>
<summary>Android</summary>

[android1-compressed.webm](https://github.com/Expensify/App/assets/44364829/fc354b53-8ef7-41c1-96de-1eab12962f46)
[android2-compressed.webm](https://github.com/Expensify/App/assets/44364829/e5352685-7230-48ad-ba1f-44521657f7aa)

</details>

JKobrynski committed:
create ReportCardLostPage, add it to navigator

JKobrynski committed:
add form to onyxkeys

JKobrynski committed:
start working on SingleOptionSelector

JKobrynski committed:
improve SingleOptionSelector style

JKobrynski committed:
add translations

JKobrynski committed:
translate SingleOptionSelectors options

JKobrynski committed:
add missing translations

JKobrynski committed:
add missing components to ReportCardLostPage

JKobrynski committed:
reorganize styles in ReportCardLostPage

JKobrynski committed:
add missing translations

JKobrynski committed:
fetch personal details in ReportCardLostPage

JKobrynski committed:
add button style prop to forms

JKobrynski committed:
improve button style

JKobrynski committed:
add comments

JKobrynski committed:
update translations

JKobrynski committed:
add spanish translations

JKobrynski committed:
adjust forms to use the latest props

JKobrynski committed:
add new form to ONYXKEYS

JKobrynski committed:
start implementing card settings api methods

JKobrynski committed:
add validation logic to report form

BeeMargarida committed:
feat: pass tag list name to the modified expense message

BeeMargarida committed:
fix: small review fixes

JKobrynski committed:
revert Accessibility change

JKobrynski committed:
revert Accessibility.md changes

JKobrynski committed:
revert Form.js changes

JKobrynski committed:
remove deprecated english translations

JKobrynski committed:
add missing prop to dependency array

JKobrynski committed:
remove unused spanish translations

JKobrynski committed:
Update CardSettings action

JKobrynski committed:
add missing translations

JKobrynski committed:
modify route definition

JKobrynski committed:
rename card actions file

JKobrynski committed:
rename onyx form key

JKobrynski committed:
allow reporting card loss from the card page

JKobrynski committed:
add translations for error message

JKobrynski committed:
finish implementing report flow

pasyukevich commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,197 @@
+    }
+
+    return (
+        <ScreenWrapper>
```
Missed `testID` on the wrapper

JKobrynski replied:
Done, thanks!

fabioh8010 commented on src/ONYXKEYS.ts:
```diff
@@ -291,6 +291,7 @@ const ONYXKEYS = {
         PRIVATE_NOTES_FORM: 'privateNotesForm',
         I_KNOW_A_TEACHER_FORM: 'iKnowTeacherForm',
         INTRO_SCHOOL_PRINCIPAL_FORM: 'introSchoolPrincipalForm',
+        REPORT_PHYSICAL_CARD: 'reportPhysicalCardLostForm',
```
```suggestion
        REPORT_PHYSICAL_CARD_FORM: 'reportPhysicalCardLostForm',
```

JKobrynski replied:
Done 😄

fabioh8010 commented on src/components/SingleOptionSelector.js:
```diff
@@ -0,0 +1,60 @@
+SingleOptionSelector.defaultProps = defaultProps;
+SingleOptionSelector.displayName = 'SingleOptionSelector';
+
+export default withLocalize(SingleOptionSelector);
```
We can use the `useLocalize` hook instead of the this HOC.

JKobrynski replied:
Unfortunately, it's not possible as of now, it causes errors. @rezkiy37 and I investigated it, and it's probably due to **Form.js** using `withLocalize` even though it's already a functional component.

JKobrynski committed:
remove buttonStyles prop from Form

JKobrynski commented on src/components/Form.js:
```diff
@@ -472,6 +477,7 @@ function Form(props) {
             props.isSubmitActionDangerous,
             props.isSubmitButtonVisible,
             props.submitButtonText,
+            props.buttonStyles,
```
Right, it wasn't necessary, I found another way to achieve the same outcome

JKobrynski committed:
rename onyx key

MrMuzyk commented on src/ONYXKEYS.ts:
```diff
@@ -291,6 +291,7 @@ const ONYXKEYS = {
         PRIVATE_NOTES_FORM: 'privateNotesForm',
         I_KNOW_A_TEACHER_FORM: 'iKnowTeacherForm',
         INTRO_SCHOOL_PRINCIPAL_FORM: 'introSchoolPrincipalForm',
+        REPORT_PHYSICAL_CARD_FORM: 'reportPhysicalCardLostForm',
```
`REPORT_PHYSICAL_CARD_LOST_FORM` ?

JKobrynski replied:
Done!

MrMuzyk commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,200 @@
+            Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
+        }
+
+        if (formData.isLoading && _.isEmpty(physicalCard.errors)) {
```
this should probably be the first `if` to return as early as possible

JKobrynski replied:
Done!

grgia commented:
Assigning @parasharrajat for C+ review

JKobrynski committed:
filter empty address values

JKobrynski committed:
make early return as early as possible

JKobrynski committed:
modify constant value

JKobrynski commented:
@parasharrajat @allroundexperts the PR is ready for review! I added screen recording for all platforms and completed the checklist

JKobrynski commented:
@allroundexperts bump 😄

allroundexperts reviewed (changes requested):
Some code refactors required.

JKobrynski committed:
remove redundant translation

JKobrynski commented on src/languages/en.ts:
```diff
@@ -815,6 +815,7 @@ export default {
         availableSpend: 'Remaining spending power',
         virtualCardNumber: 'Virtual card number',
         physicalCardNumber: 'Physical card number',
+        reportCardLostOrDamaged: 'Report physical card loss / damage',
```
Removed!

JKobrynski commented on src/languages/es.ts:
```diff
@@ -810,6 +810,7 @@ export default {
         availableSpend: 'Capacidad de gasto restante',
         virtualCardNumber: 'Número de la tarjeta virtual',
         physicalCardNumber: 'Número de la tarjeta física',
+        reportCardLostOrDamaged: 'Notificar la pérdida/daño de la tarjeta física',
```
Removed!

JKobrynski committed:
use correct form key

JKobrynski commented on src/libs/actions/Card.js:
```diff
@@ -0,0 +1,52 @@
+            successData: [
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: ONYXKEYS.FORMS.REPORT_FRAUD_FORM,
```
No reason at all, nice catch!

JKobrynski committed:
create findPhysicalCard helper funciton

JKobrynski commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,200 @@
+    usePrivatePersonalDetails();
+
+    const domainCards = CardUtils.getDomainCards(cardList)[domain];
+    const physicalCard = _.find(domainCards, (card) => !card.isVirtual) || {};
```
Done 😄

JKobrynski commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,200 @@
+        }
+    };
+
+    const handleOptionSelect = (option) => {
+        setReason(option);
+    };
```
If you take a look at the `onSubmit` function, it passes the `reason` to `requestReplacementExpensifyCard` method, does that answer your question?

JKobrynski committed:
split onSubmit into two functions

JKobrynski commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,200 @@
+        return <NotFoundPage />;
+    }
+
+    const onSubmit = () => {
+        if (!isReasonConfirmed) {
+            setIsReasonConfirmed(true);
+
+            setShouldShowAddressError(false);
+
+            return;
+        }
+
+        if (!formattedAddress) {
+            setShouldShowAddressError(true);
+            return;
+        }
+
+        if (formattedAddress) {
+            CardActions.requestReplacementExpensifyCard(physicalCard.cardID, reason);
+        }
+    };
```
I split it into two separate methods depending on step, let me know what you think

JKobrynski committed:
remove unused funciton

JKobrynski committed:
split use effect into two simpler ones

JKobrynski commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,200 @@
+        }
+
+        FormActions.setErrors(ONYXKEYS.FORMS.REPORT_PHYSICAL_CARD_FORM, physicalCard.errors);
+    }, [domain, formData.isLoading, prevIsLoading, physicalCard.errors]);
```
I simplified it a bit and split into two smaller effects, let me know your thoughts 😄

JKobrynski committed:
move getFormattedAddress method to PersonalDetailsUtils

JKobrynski commented on src/libs/CardUtils.ts:
```diff
@@ -56,4 +56,27 @@ function maskCard(lastFour = ''): string {
+    const formatted = addressItems.join(', ');
+
+    // Remove the last comma of the address
+    return formatted.trim().replace(/,$/, '');
```
Yes, we have something similar in `PersonalDetailsInitialPage`. Take a look at [this PR](https://github.com/Expensify/App/pull/26778), where this method was modified and made reusable (and approved). I used the same solution for this PR, let me know what you think!

JKobrynski commented on src/libs/CardUtils.ts:
```diff
@@ -56,4 +56,27 @@ function maskCard(lastFour = ''): string {
+    const formatted = addressItems.join(', ');
+
+    // Remove the last comma of the address
+    return formatted.trim().replace(/,$/, '');
```
I removed this method, so I guess it's outdated 😄

JKobrynski commented on src/libs/CardUtils.ts:
```diff
@@ -56,4 +56,27 @@ function maskCard(lastFour = ''): string {
+ * Formats an address object into an easily readable string
+ *
+ * @returns - formatted address
+ */
```
Method removed (explained in other comments)

JKobrynski commented on src/components/SingleOptionSelector.js:
```diff
@@ -0,0 +1,60 @@
+import CONST from '../CONST';
+import Text from './Text';
+import PressableWithoutFeedback from './Pressable/PressableWithoutFeedback';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
```
Unfortunately it's not possible, it causes an error (discussed in [this comment](https://github.com/Expensify/App/pull/28141#discussion_r1340921737)). It's probably caused by `Form` that uses `withLocalize` and wraps `SingleOptionSelector` in that case.

marcaaron replied:
I think we can address this in a follow up and this is not a blocker for now.

JKobrynski commented:
@allroundexperts PR updated!

JKobrynski committed:
add early returns to effects

JKobrynski commented:
@allroundexperts lint issues fixed!

allroundexperts reviewed (changes requested):
Changes needed!

JKobrynski committed:
minor code changes, improvements

JKobrynski commented on src/pages/settings/Wallet/ReportCardLostPage.js:
```diff
@@ -0,0 +1,202 @@
+
+        if (prevIsLoading && !formData.isLoading) {
+            Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
+        }
```
Updated, however I had to do it that way:

```javascript
if (!_.isEmpty(physicalCard.errors) || !(prevIsLoading && !formData.isLoading)) {
    return;
}

Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute(domain));
```

JKobrynski committed:
remove left margin from singleOptionSelector

JKobrynski committed:
remove preselected reason, add reason error

JKobrynski committed:
shrink pressable area in SingleOptionSelector

JKobrynski committed:
allow going back to the previous step

JKobrynski committed:
use FormAlertWithSubmitButton instead of Form

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski commented:
@allroundexperts thanks for so many useful comments! I made many changes to address them, so here is a quick summary:

1. Simplified `useEffect` in `ReportCardLostPage`
2. Edited `Card.js` to reset error in optimistic data
3. As for the titles that might have wrong font size/family, I used Figma to check the details (they might be outdated) and it's `Expensify New Kansas`, 22px exactly like in the app
4. There is no default selected option in `SingleOptionSelector` (nothing is selected by default)
5. Added error handling for the reason step (no selected reason - show error)
6. Removed left margin from `SingleOptionSelector` (it was added by mistake, was never in designs)
7. Made only text + `SelectCircle` clickable in `SingleOptionSelector` instead of the whole row
8. Improved back button press handler, to allow users to go back to the previous step
9. Fixed the hover on address `MenuItem`
10. The console error must've been related to something else, I merged the latest main branch and it disappeared for me

Additionally:

I also worked on [add “Report virtual card fraud” page and route](https://github.com/Expensify/App/pull/28312) and I thought [this comment](https://github.com/Expensify/App/pull/28312#discussion_r1345031981) also applies to this PR, so I removed `Form` and used `FormAlertWithSubmitButton` directly, let me know what you think!

shawnborton commented:
> Not sure if this margin left was in the designs. @shawnborton to confirm this..

Agree, no need for left margin.

So far the headline font style looks correct to me. The only minor feedback is that I think we want 12px of padding below the page header bar and above the headline.

For this screen here:
![image](https://github.com/Expensify/App/assets/2319350/ce27de4f-ed42-44d7-8f02-ada0613cdf48)


I think we want the description/paragraph text to appear above the push to page row. And I agree with your comments that it should stretch full edge-to-edge.

shawnborton commented:
For this component here:
<img width="376" alt="CleanShot 2023-10-06 at 07 58 34@2x" src="https://github.com/Expensify/App/assets/2319350/99379fac-ab1c-43f3-943a-561e0cd0dc25">


Why isn't this using our standard option row component? It should have bold text and slightly different alignment. Let's reuse existing components here if we can.

JKobrynski committed:
remove unnecessary comment

JKobrynski committed:
add extra padding on top

JKobrynski committed:
use MenuItem instead of MenuItemWithTopDescription

JKobrynski commented:
@shawnborton added extra padding on top, used the existing component you asked for 😄

shawnborton commented:
Awesome, are the screenshots updated too?

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski commented:
@allroundexperts @shawnborton conflicts resolved, screen recordings updated!

shawnborton commented:
Nice, this feels pretty good to me!

JKobrynski committed:
support multiline title

JKobrynski commented:
@allroundexperts I updated text styles, it's now fixed 😄 (attaching a screenshot). About that transition, I don't think it was required (unless that changed) I even asked @grgia about it in [this thread](https://expensify.slack.com/archives/C027726KHT4/p1694519993227079), and she confirmed it can be one screen, let me know what you think  though!

![Screenshot_1696923479](https://github.com/Expensify/App/assets/44364829/b09fcd73-50cc-46d4-9630-18643367b4f8)

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski committed:
fix linting

marcaaron commented on src/ONYXKEYS.ts:
```diff
@@ -291,6 +291,7 @@ const ONYXKEYS = {
         PRIVATE_NOTES_FORM: 'privateNotesForm',
         I_KNOW_A_TEACHER_FORM: 'iKnowTeacherForm',
         INTRO_SCHOOL_PRINCIPAL_FORM: 'introSchoolPrincipalForm',
+        REPORT_PHYSICAL_CARD_FORM: 'reportPhysicalCardForm',
```
This should be `requestPhysicalCardForm`

JKobrynski replied:
Done!

marcaaron reviewed (changes requested):
Changes look good. Similar to https://github.com/Expensify/App/pull/28312 we will want to do internal QA on the changes.

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski committed:
change forms onyx key

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski commented:
@allroundexperts conflicts resolved

joelbettner commented:
Just a heads up, it looks like the testing of this PR created this issue: https://github.com/Expensify/Expensify/issues/325564

`RequestReplacementExpensifyCard` was called without a `cardID`.

allroundexperts reviewed (approved):
Looks good!

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski committed:
modify route name to singular

grgia commented on src/components/SelectCircle.js:
```diff
@@ -9,15 +9,20 @@ import themeColors from '../styles/themes/default';
 function SelectCircle(props) {
     return (
-        <View style={[styles.selectCircle, styles.alignSelfCenter]}>
+        <View style={[styles.selectCircle, styles.alignSelfCenter, props.styles]}>
```
Shouldn't props.styles be spread since it's an array?

JKobrynski replied:
Done

grgia reviewed (commented):
minor comment

JKobrynski committed:
spread styles array

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

grgia reviewed (commented):
<img width="211" alt="image" src="https://github.com/Expensify/App/assets/38015950/9baa70f4-e4e1-49ca-bfa0-9a93a76053a5">

![image](https://github.com/Expensify/App/assets/38015950/fbc605f8-fd86-4995-a778-edd6e72d8292)

Looks like the border color for the circle toggle is off- it should be our `icons` color

grgia reviewed (commented):
@JKobrynski can you confirm this test

If reporting as lost/stolen, then physical card is removed from card page

![image](https://github.com/Expensify/App/assets/38015950/d6ee56c3-2055-4590-b105-1213c83d1731)


If reporting as damaged, then physical card remains on card page

![image](https://github.com/Expensify/App/assets/38015950/ef33f9af-b411-4bd7-9909-d002a4cb5ecf)

JKobrynski committed:
Merge branch 'main' into addReportCardLostOrDamaged

JKobrynski committed:
change SelectCircle border color

JKobrynski commented:
@grgia border color changed! Regarding the second comment: yes, that's how I understand it based on the google doc, however we established that this PR won't implement full backend functionality (just boilerplate) as I am not able to test this on my end, and I don't even know if that API method is implemented yet, hope that answers your question!

grgia commented:
Will let @marcaaron take a look before merging so I don't dismiss his review

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6567842419) to staging by https://github.com/marcaaron in version: 1.3.87-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6592612183) to production by https://github.com/francoisl in version: 1.3.87-12 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6592629328) to staging by https://github.com/marcaaron in version: 1.3.88-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6608779761) to production by https://github.com/chiragsalian in version: 1.3.88-11 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅
